detectors/node/opentelemetry-resource-detector-aws/src/detectors/AwsEcsDetector.ts

```diff
@@ -38,27 +41,65 @@ import { getEnv } from '@opentelemetry/core';
 export class AwsEcsDetector implements Detector {
   readonly CONTAINER_ID_LENGTH = 64;
   readonly DEFAULT_CGROUP_PATH = '/proc/self/cgroup';
+
   private static readFileAsync = util.promisify(fs.readFile);
 
+  private static async getUrlAsJson(url: string): Promise<any> {
+    return new Promise<string>((resolve, reject) => {
+      const request = http.get(url, (response: any) => {
+        if (response.statusCode >= 400) {
+          reject(`Request to '${url}' failed with status ${response.statusCode}`);
+        }
+        /*
+        * Concatenate the response out of chunks:
+        * https://nodejs.org/api/stream.html#stream_event_data
+        */
+        let responseBody = '';
+        response.on('data', (chunk: Buffer) => (responseBody += chunk.toString()));
+        // All the data has been read, resolve the Promise
+        response.on('end', () => resolve(responseBody));
```

Comment:
If you look at the event ordering for http requests [1] I think it is possible for the response to not `end` but instead `error` in which case this promise will never resolve or reject. Look at the section that says "In the case of a premature connection close after the response is received"

[1]: https://nodejs.org/api/http.html#httprequesturl-options-callback

Reply:
I thought this would be covered by the `request.on('error', reject);` a few lines below @dyladan

Reply:
Actually no, you are right, brb

Reply:
Done in 30b8708b